sdk/examples/xo_python/sawtooth_xo/xo_message_factory.py

```diff
@@ -13,57 +13,65 @@
 # limitations under the License.
 # ------------------------------------------------------------------------------
 
+import hashlib
+
 from sawtooth_processor_test.message_factory import MessageFactory
 
 
+# namespace
+FAMILY_NAME = 'xo'
+XO_NAMESPACE = hashlib.sha512(FAMILY_NAME.encode('utf-8')).hexdigest()[:6]
+
+def make_xo_address(name):
+    return XO_NAMESPACE + hashlib.sha512(name.encode('utf-8')).hexdigest()[-64:]
+
+# encodings
+def encode_txn_payload(action, name, space=''):
+    return ','.join([str(data) for data in (action, name, space)]).encode()
+
+
 class XoMessageFactory:
```

Comment:
I can do that in this PR, or open another PR to do it for all the TPs -- intkey and registry follow this same pattern, while config has a different but still odd arrangement.